src/testnet/helium/event_dispatcher.rs

```diff
@@ -0,0 +1,242 @@
+                    "stx_mint_event": event_data
+                }),
+                StacksTransactionEvent::STXEvent(STXEventType::STXBurnEvent(event_data)) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "stx_burn_event",
+                    "stx_burn_event": event_data
+                }),
+                StacksTransactionEvent::NFTEvent(NFTEventType::NFTTransferEvent(event_data)) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "nft_transfer_event",
+                    "nft_transfer_event": event_data
+                }),
+                StacksTransactionEvent::NFTEvent(NFTEventType::NFTMintEvent(event_data)) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "nft_mint_event",
+                    "nft_mint_event": event_data
+                }),
+                StacksTransactionEvent::FTEvent(FTEventType::FTTransferEvent(event_data)) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "ft_transfer_event",
+                    "ft_transfer_event": event_data
+                }),
+                StacksTransactionEvent::FTEvent(FTEventType::FTMintEvent(event_data)) => json!({
+                    "txid": format!("0x{:?}", txid),
+                    "type": "ft_mint_event",
+                    "ft_mint_event": event_data
+                }),
+            }).collect();
+
+        let serialized_txs: Vec<serde_json::Value> = receipts.iter().map(|artifact| {
+            
+            let response_data = match &artifact.result {
+                Value::Response(data) => data,
+                _ => unreachable!(),
+            };
+
+            let tx = &artifact.transaction;
+            let mut unsigned_tx_bytes = vec![];
+            tx.consensus_serialize(&mut unsigned_tx_bytes).unwrap();
+            let formatted_bytes: Vec<String> = unsigned_tx_bytes.iter().map(|b| format!("{:02x}", b)).collect();
+            json!({
+                "txid": format!("0x{}", tx.txid()),
+                "success": response_data.committed,
+                "result": response_data.data,
+                "raw_tx": format!("0x{}", formatted_bytes.join("")),
+            })
+        }).collect();
+        
+        // Wrap events
+        let payload = json!({
+            "block_hash": format!("0x{:?}", chain_tip.block_hash()),
+            "block_height": chain_tip_info.block_height,
+            "index_block_hash": format!("0x{:?}", chain_tip_info.index_block_hash()),
+            "parent_block_hash": format!("0x{:?}", chain_tip.header.parent_block),
+            "parent_microblock": format!("0x{:?}", chain_tip.header.parent_microblock),
+            "events": serialized_events,
+            "transactions": serialized_txs,
+        }).to_string();
+
+        // Send payload
+        let _res = stream.write_bufs(&vec![payload.as_bytes().into()]);
```

Comment:
What happens if the `write_bufs` fails?  Is it logged anywhere?

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1319/commits/981385ca7f1e5f515c9cdb440e8704847ec6d1a4